frontend/public/components/modals/disable-application-modal.tsx

```diff
@@ -2,71 +2,62 @@ import * as React from 'react';
 import * as _ from 'lodash-es';
 
 import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
-import { PromiseComponent, history, resourceListPathFromModel } from '../utils';
+import { history, resourceListPathFromModel, withHandlePromise } from '../utils';
 import { ClusterServiceVersionKind, SubscriptionKind } from '../operator-lifecycle-manager';
 import { K8sKind, K8sResourceKind } from '../../module/k8s';
 import { ClusterServiceVersionModel, SubscriptionModel } from '../../models';
+import { getActiveNamespace } from '../../actions/ui';
 
-export class DisableApplicationModal extends PromiseComponent<DisableApplicationModalProps, DisableApplicationModalState> {
-  public state: DisableApplicationModalState;
+export const DisableApplicationModal = withHandlePromise((props: DisableApplicationModalProps) => {
+  const [deleteCSV, setDeleteCSV] = React.useState(true);
 
-  constructor(public props: DisableApplicationModalProps) {
-    super(props);
-    this.state.deleteCSV = true;
-  }
-
-  private submit(event): void {
+  const submit = (event) => {
     event.preventDefault();
 
-    const {subscription, k8sKill} = this.props;
+    const {subscription, k8sKill} = props;
     const deleteOptions = {kind: 'DeleteOptions', apiVersion: 'v1', propagationPolicy: 'Foreground'};
     const promises = [k8sKill(SubscriptionModel, subscription, {}, deleteOptions)]
-      .concat(_.get(this.props.subscription, 'status.installedCSV') && this.state.deleteCSV
+      .concat(_.get(subscription, 'status.installedCSV') && deleteCSV
         ? k8sKill(ClusterServiceVersionModel, {metadata: {name: subscription.status.installedCSV, namespace: subscription.metadata.namespace}} as ClusterServiceVersionKind, {}, deleteOptions).catch(() => Promise.resolve())
         : []);
 
-    this.handlePromise(Promise.all(promises)).then(() => {
-      this.props.close();
-      history.push(resourceListPathFromModel(SubscriptionModel, subscription.metadata.namespace));
+    props.handlePromise(Promise.all(promises)).then(() => {
+      props.close();
+      history.push(resourceListPathFromModel(SubscriptionModel, getActiveNamespace()));
```

Comment:
Do we need to make a similar change to `delete-modal.jsx`?

https://github.com/openshift/console/blob/master/frontend/public/components/modals/delete-modal.jsx#L37

Reply:
Good idea, switched to use that logic (saves us an unnecessary page reload as well).